util/network-alert/src/alert_relayer.rs

```diff
@@ -0,0 +1,135 @@
+//! AlertRelayer
+//! We implment a Bitcoin like alert system, n of m alert key holders can decide to send alert
+//messages to all client
+//! to leave a space to reach consensus offline under critical bugs
+//!
+//! A cli to generate alert message,
+//! A config option to set alert messages to broard cast.
+//
+use crate::config::Config;
+use crate::notifier::Notifier;
+use crate::verifier::Verifier;
+use crate::BAD_MESSAGE_BAN_TIME;
+use ckb_core::alert::Alert;
+use ckb_network::{CKBProtocolContext, CKBProtocolHandler, PeerIndex, TargetSession};
+use ckb_protocol::{get_root, AlertMessage};
+use ckb_util::Mutex;
+use flatbuffers::FlatBufferBuilder;
+use fnv::FnvHashSet;
+use log::{debug, info, trace};
+use lru_cache::LruCache;
+use std::convert::TryInto;
+use std::sync::Arc;
+
+const KNOWN_LIST_SIZE: usize = 64;
+
+/// AlertRelayer
+/// relay alert messages
+#[derive(Clone)]
+pub struct AlertRelayer {
+    notifier: Arc<Mutex<Notifier>>,
+    verifier: Arc<Verifier>,
+    known_lists: LruCache<PeerIndex, FnvHashSet<u32>>,
+}
+
+impl AlertRelayer {
+    pub fn new(client_version: String, config: Config) -> Self {
+        AlertRelayer {
+            notifier: Arc::new(Mutex::new(Notifier::new(client_version))),
+            verifier: Arc::new(Verifier::new(config)),
+            known_lists: LruCache::new(KNOWN_LIST_SIZE),
+        }
+    }
+
+    pub fn notifier(&self) -> Arc<Mutex<Notifier>> {
+        Arc::clone(&self.notifier)
+    }
+
+    pub fn verifier(&self) -> Arc<Verifier> {
+        Arc::clone(&self.verifier)
```

Comment:
Avoid Arc clone implicitly.

Reply:
Why? This there some best practice recommendations?

Reply:
Sorry that I don't know any best practices around it.
I just feel like this style since https://github.com/nervosnetwork/ckb/pull/653#discussion_r281888622

Reply:
Ok, make sense to me, I' ll update it.